pkg/provenance.go

```diff
@@ -642,17 +656,57 @@ func getBaseRef(environment map[string]interface{}) (string, error) {
 		return "", nil
 	}
 
-	eventPayload, ok := environment["github_event_payload"]
+	payload, err := getEventPayload(environment)
+	if err != nil {
+		return "", err
+	}
+
+	return getAsString(payload, "base_ref")
+}
+
+func getTargetCommittish(environment map[string]interface{}) (string, error) {
+	eventName, err := getAsString(environment, "github_event_name")
+	if err != nil {
+		return "", err
+	}
+
+	if eventName != "release" {
+		return "", nil
+	}
+
+	payload, err := getEventPayload(environment)
+	if err != nil {
+		return "", err
+	}
+
+	// For a release event, we look for release.target_commitish.
+	releasePayload, ok := payload["release"]
 	if !ok {
-		return "", fmt.Errorf("%w: %s", ErrorInvalidDssePayload, "parameters type event payload")
+		return "", fmt.Errorf("%w: %s", ErrorInvalidDssePayload, "release absent from payload")
 	}
 
-	payload, ok := eventPayload.(map[string]interface{})
+	release, ok := releasePayload.(map[string]interface{})
 	if !ok {
-		return "", fmt.Errorf("%w: %s", ErrorInvalidDssePayload, "parameters type payload")
+		return "", fmt.Errorf("%w: %s", ErrorInvalidDssePayload, "parameters type releasePayload")
 	}
 
-	return getAsString(payload, "base_ref")
+	branch, err := getAsString(release, "target_commitish")
+	if err != nil {
+		return "", fmt.Errorf("%w: %s", err, "target_commitish not present")
```

Comment:
target_committish

Reply:
the payload has a `target_commitish` field, but not `target_committish`. Not sure if it's intentional or a typo.

Reply:
Ah, I see. It's actually a typo in the payload. Ok.